src/libsystemd/sd-journal/journal-file.c

```diff
@@ -92,6 +92,19 @@
 #  pragma GCC diagnostic ignored "-Waddress-of-packed-member"
 #endif
 
+_const_ static int mmap_prot_from_open_flags(int flags) {
+        switch (flags & O_ACCMODE) {
+        case O_RDONLY:
+                return PROT_READ;
+        case O_WRONLY:
+                return PROT_WRITE;
+        case O_RDWR:
+                return PROT_READ|PROT_WRITE;
+        default:
+                assert_not_reached();
+        }
+}
```

Comment:
this function is *super* generic btw, I'd probably keep it in the shared path, it sounds like this is what we want sooner or later for any mmap() code...

Reply:
Hmm, it has exactly one user, added in 2011. mmap is not a terribly nice interface to use, and I think it's quite likely that we'll never add any more uses. But if that happens, it should be trivial to move the function to some public header… I don't think we need to do this proactively for an uncertain future use case.